src/value_generators/include/value_generators/DocumentGenerator.hpp

```diff
@@ -182,18 +246,45 @@ class Expression {
     virtual Value evaluate(genny::DefaultRandom& rng) const = 0;
 };
 
+template <class T>
+class TypedExpression {
+    UniqueExpression _expression;
```

Comment:
[style] Explicitly declare a private: section and move these lines after the public: section?